flink-table-store-hive/flink-table-store-hive-connector/src/main/java/org/apache/flink/table/store/mapred/TableStoreRecordReader.java

```diff
@@ -36,24 +41,44 @@ public class TableStoreRecordReader implements RecordReader<Void, RowDataContain
     private final RecordReaderIterator<RowData> iterator;
     private final long splitLength;
 
+    @Nullable private final ProjectedRowData reusedProjectedRow;
+
     private float progress;
 
     public TableStoreRecordReader(
-            org.apache.flink.table.store.file.utils.RecordReader<RowData> wrapped,
-            long splitLength) {
-        this.iterator = new RecordReaderIterator<>(wrapped);
-        this.splitLength = splitLength;
+            TableRead read,
+            TableStoreInputSplit split,
+            List<String> columnNames,
+            List<String> selectedColumns)
```

Comment:
Add documentation here: Still output the original schema of the Row, the role of `selectedColumns` is to read out the required fields, other fields to fill Nulls.